pluginlib/cmake/pluginlib_enable_plugin_testing.cmake

```diff
@@ -0,0 +1,223 @@
+    set(ARG_PACKAGE_NAME "${PROJECT_NAME}")
+  endif()
+  if(NOT ARG_PACKAGE_XML)
+    set(ARG_PACKAGE_XML "${CMAKE_SOURCE_DIR}/package.xml")
+  endif()
+  if(NOT ARG_PLUGIN_CATEGORY)
+    message(FATAL_ERROR "PLUGIN_CATEGORY is a required argument")
+  endif()
+  if(NOT ARG_PLUGIN_DESCRIPTIONS)
+    message(FATAL_ERROR "At least one plugin description via PLUGIN_DESCRIPTIONS is required")
+  endif()
+  if(NOT ARG_PLUGIN_LIBRARIES)
+    message(FATAL_ERROR "At least one plugin library target via PLUGIN_LIBRARIES is required")
+  endif()
+
+  set(target_name "pluginlib_enable_plugin_testing__${ARG_PLUGIN_CATEGORY}__${ARG_PACKAGE_NAME}")
+  if(TARGET "${target_name}")
+    message(FATAL_ERROR "pluginlib_enable_plugin_testing has already been called with category '${ARG_PLUGIN_CATEGORY}' and package name '${ARG_PACKAGE_NAME}'")
+  endif()
+
+  #####
+  # Plan out mock install space before configuring things to create it
+  #####
+
+  # Lists of equal size
+  # If input_files is not "", then copy that to the path in "output_files"
+  # If input_files is "", then write string from input_content to the path in "output_files"
+  # FOOBAR is a workaround to enable appending empty elements when the list is empty
+  set(input_content "FOOBAR")
+  set(input_files "FOOBAR")
+  set(output_files "FOOBAR")
+
+  # Each "PACKAGE_NAME" goes to it's own folder
+  set(prefix "${CMAKE_CURRENT_BINARY_DIR}/pluginlib_enable_plugin_testing/install/${ARG_PACKAGE_NAME}__${ARG_PLUGIN_CATEGORY}")
+
+  # Install package.xml, renaming to just `package.xml` if needed
+  set(fake_install_dir "${prefix}/share/${ARG_PACKAGE_NAME}/")
+  list(APPEND input_files "${ARG_PACKAGE_XML}")
+  list(APPEND input_content "")
+  list(APPEND output_files "${fake_install_dir}/package.xml")
+
+  # Add blank file indicating existance of package in the ament index
+  list(APPEND input_files "")
+  list(APPEND input_content "")
+  list(APPEND output_files "${prefix}/share/ament_index/resource_index/packages/${ARG_PACKAGE_NAME}")
+
+  # Write the plugin descriptions in share/
+  set(ament_index_plugin_content "")
+  foreach(plugin_description_path "${ARG_PLUGIN_DESCRIPTIONS}")
+    list(APPEND input_files "${plugin_description_path}")
+    list(APPEND input_content "")
+    list(APPEND output_files "${prefix}/share/${ARG_PACKAGE_NAME}/${plugin_description_path}")
+
+    set(ament_index_plugin_content "${ament_index_plugin_content}\nshare/${ARG_PACKAGE_NAME}/${plugin_description_path}")
+  endforeach()
+  # Get rid of leading newline
+  string(STRIP "${ament_index_plugin_content}" ament_index_plugin_content)
+
+  # Write one file that says where all of this package's plugin description files are
+  list(APPEND input_files "")
+  list(APPEND input_content "${ament_index_plugin_content}")
+  list(APPEND output_files "${prefix}/share/ament_index/resource_index/${ARG_PLUGIN_CATEGORY}__pluginlib__plugin/${ARG_PACKAGE_NAME}")
+
+  # Get rid of FOOBAR
+  list(POP_FRONT input_files)
```

Comment:
This requires CMake 3.15.7 
https://cmake.org/cmake/help/v3.15/command/list.html

Before that, `POP_FRONT` is not available as a list subcommand in cmake.

Reply:
That also breaks the build on my ubuntu 18.04, as its cmake version there is 3.10

Reply:
Fixed by https://github.com/ros/pluginlib/pull/203